GSA-TTS/FAC
# Revise/Collapse Checklist & HTML Refactor

# Revise/Collapse Checklist & HTML Refactor

Issue: https://github.com/GSA-TTS/FAC/issues/3161

## Changes:
1. After locking for certification, the checklist looks pretty different.
    * The form and upload pages are hidden away.
    * There's new key info text explaining the lock/unlock and preview downloads. This includes text about email notifications.
    * There are button links to the downloads and the unlock page.
    * The unlock button is not available after submission has been completed
2. HTML refactor:
    * Many checklist components are now in their own files. Notably, the different key info sections have their own files.
    * The section block has moved into the same directory as the other checklist components. The template tag has been updated to match.
    * The grand and glorious `submission_checklist.html` is much smaller due to these changes, and you can see the important logic on _one screen_. How crazy is that?
    * A bunch of comment updates that are probably/maybe useful. 
3. Minor things:
    * The download buttons on the actual certification screens are gone, in favor of the ones in the checklist.
    * 403's that come from accessing a cert screen without access now show why. Instead of "bad role", we have "Bad Role. User is not the Auditee/or Certifying Official". 

## How to Test:
1. Switch to this branch and run normally.
2. Go through a submission.
3. Ensure that things appear the same, until after locking for certification. 
4. After locking, ensure that the new form of the checklist:
    * Has the new key info text
    * Has new download & unlock buttons
    * Has _functional_ download and unlock buttons
5. Step through to submission. Make sure nothing breaks. 

As a bonus, you can run Cypress to verify that these changes are non-breaking for the full-submission tests. 

## Screenshots:
New checklist form:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/85d19a12-3ba5-4250-b86f-ad4ad4daf8e1)

Post submission/acceptance, the unlock goes away:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/362f6ac7-1d29-46df-9fe9-3a487ac8274e)

403 display change:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/3d86280e-2cc2-43be-8cf7-92d128d5a2e7)

Removed the download buttons from the cert pages:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/bbd0d0f4-af2f-4238-9192-e36a68195a93)

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
